bite-size-notes/week-3-e-optimal-control-theory.md

```diff
@@ -0,0 +1,210 @@
+
+where $K(x)$ is the terminal cost for all states $x$. The Boundary condition is a consequence of our specific problem formulation.
+
+In simpler terms, we can say that for a given time interval $[t_0,t_1)$ and state $x$, the optimal value function $V(t,x)$ satisfies a recursive relationship, that is, for  any time $t$ within the interval and any small time increment $\Delta t$, the value function $V(t,x)$ can be expressed as the sum of the cost incurred over the current interval and the optimal cost-to-go from the end of the interval.
+
+## Main Body
+
+An optimal control problem involves finding control inputs that optimize a certain criterion while satisfying system dynamics and constraints. The problem formulation of an optimal control problem typically requires:
+1. ***A mathematical description (or model) of the process to be controlled***
+   - This component involves defining the dynamics of the system to be controlled. The dynamics function $f(x_k, u_k, k)$ describes how the state of the system evolves over time in response to control inputs.
+2. ***A statement of the physical constraints***
+   - The physical constraints on the system, represented by sets $\mathcal{X}$ and $\mathcal{U}$, define the feasible state and control spaces, respectively. These constraints ensure that the system operates within permissible bounds.
+   - Additionally, equality constraints $g_i(x_k, u_k) = 0$ and inequality constraints $h_i(x_k, u_k) \leq 0$ may be imposed to further restrict the feasible space.
+3. ***Specifications of the performance criterion***
+   - The goal of the optimal control problem is to optimize a performance criterion, which is typically represented by a cost functional. The cost functional $J(x_k, u_k, k)$ evaluates the quality of a control trajectory over a given time horizon.
+   - The objective is to minimize the total cost, which consists of both the running cost $$\sum_{k=0}^{K} J(x_k, u_k, k)$$ over the entire trajectory and the terminal cost $J_{\text{term}}(x_{K+1})$ at the final state.
+  
+#### General Setup of Optimal Control Problem
+The general Setup of Optimal Control Problem, in discrete time and finite horizon:
+
+$$\begin{aligned}
+& \min_{\begin{aligned}[t]
+& u_0, u_1, \ldots, u_K \\
+& x_0, x_1, \ldots, x_{K+1}
+\end{aligned}} && \sum_{k=0}^{K} J(x_k, u_k, k) + J_{\text{term}}(x_{K+1}) \\
+& \text{s.t.} && x_{k+1} = f(x_k,u_k,k) \quad (k = 0,1,\ldots, K) \\
+&&& x_k \in \mathcal{X}, \quad u_k \in \mathcal{U}, \quad x_0 = x_{\text{current}} \\
+&&& g_i(x_k,u_k) = 0 \quad (i = 1,\ldots, G) \\
+&&& h_i(x_k,u_k) \leq 0 \quad (i = 1, \ldots, H)
+\end{aligned}$$
+
+where $J_{\text{term}}(x_{K+1})$ is the terminal cost,
+      $$\sum_{k=0}^{K} J(x_k, u_k, k)$$ is the summation of the running cost,
+      $x_{k+1} = f(x_k,u_k,k) \quad (k = 0,1,\ldots, K)$ is the dynamics of the system,
+      $x_k \in \mathcal{X}, \quad u_k \in \mathcal{U}, \quad x_0 = x_{\text{current}}$ are the system constraints,
```

Comment:
Do you think expressing x0 as initial state instead of current state better? Since from where I stand,  x_k in the discrete time can be any current time.